kpops/components/base_components/pipeline_component.py

```diff
@@ -67,6 +71,30 @@ def __init__(self, **kwargs) -> None:
         self.set_input_topics()
         self.set_output_topics()
 
+    @property
+    def input_topics(self) -> list[str]:
+        """Get all the input topics from config."""
+        return []
+
+    @property
+    def extra_input_topics(self) -> dict[str, list[str]]:
+        """Get extra input topics list from config."""
+        return {}
+
+    @property
+    def output_topic(self) -> str | None:
+        """Get output topic from config."""
+        return None
+
+    @property
+    def extra_output_topics(self) -> dict[str, str]:
+        """Get extra output topics list from config."""
+        return {}
+
+    @property
+    def id(self) -> str:
+        return f"component-{self.full_name}"
```

Comment:
do we still need the prefix?

Reply:
yes! this is because topic and component can have the same name.